libraries/AP_BattMonitor/AP_BattMonitor_Backend.h

```diff
@@ -114,3 +118,23 @@ class AP_BattMonitor_Backend
     float       _resistance_voltage_ref; // voltage used for maximum resistance calculation
     float       _resistance_current_ref; // current used for maximum resistance calculation
 };
+
+#if AP_BATTERY_SCRIPTING_ENABLED
+struct BattMonitorScript_State {
+    float voltage; // Battery voltage in volts
+    bool healthy; // True if communicating properly
+    uint8_t cell_count; // Number of valid cells in state
+    uint8_t capacity_remaining_pct; // Remaining battery capacity in percent, 255 for invalid
+    uint16_t cell_voltages[32]; // allow script to have up to 32 cells, will be limited internally
+    uint16_t cycle_count; // Battery cycle count, 255 for unavailable
```

Comment:
Surly this should be a magic value of 65535?

Reply:
fixed